oauth-browser-based-apps.md

```diff
@@ -250,12 +252,14 @@ in using OAuth even in a common-domain architecture:
 * Allows more flexibility in the future, such as if you were to later add a new domain to the system. With OAuth already in place, adding a new domain wouldn't require any additional rearchitecting.
 * Being able to take advantage of existing library support rather than writing bespoke code for the integration. 
 * Centralizing login and multifactor support, account management, and recovery at the OAuth server, rather than making it part of the application logic.
+* Splitting of responsibilities between authenticating a user and serving resources 
 
-Using OAuth for browser-based apps in a first-party same-domain scenario provides these advantages, and can be accomplished by either of the two architectural patterns described below.
+Using OAuth for browser-based apps in a first-party same-domain scenario provides these advantages, and can be accomplished by any of the architectural patterns described below.
 
+<!-- Changed the wording here to try to be consistent and call it BFF everywhere (backend/application server/BFF where used interchangeably -->
 
-JavaScript Applications with a Backend
---------------------------------------
+JavaScript Applications with a stateful BFF
```

Comment:
I think we should avoid using the term "stateful" here since that's not actually a relevant property of this architecture. It's possible to make the BFF stateless and use encrypted cookies to store all the session data and things still work as described in this pattern.

Reply:
The relevant property that I wanted to underline was the fact that some state needed to be kept, and that it needed to be inaccessible from the frontend. A kind of viewstate solution would indeed work too. 

My point was that this wasn't just any backend: SPA with stateless backends are very common nowadays, in contrast to stateful backends or viewstate approaches. This makes big difference though: this implies some infrastructure requirements that just aren't needed in other solutions (local/session storage or service workers).

 I'll try to find some better wording.

Reply:
After some reflection: I tried to remove that "stateful" aspect, but I'm actually quite convinced that this property is indeed characteristic of this architecture: even if that state can be stored in the frontend (in cookies or otherwise), which typically helps for scaling across multiple backend nodes, there is still backend state that needs to be maintained.

My suggestion is to keep the "stateful" wording, maybe with a word of clarification.

Do you have another proposal or view about how to describe the architecture in the best possible way?